### DISCUSSION TITLE
RemoteConfig supports property wrappers for Swift 5.1 or later

### DISCUSSION THREAD

**Discussion by fumito-ito** [firebase/firebase-ios-sdk#9828] [Pitches]
Property wrapper is the feature available for Swift 5.1 or later.

> A property wrapper adds a layer of separation between code that manages how a property is stored and the code that defines a property. 

This function has great compatibility with Remote Config. In my use case , it will be able to use fetching feature flags.

```swift
struct FeatureFlags {
    @RemoteConfig(forKey: "is_new_feature_available")
    var isNewFeatureAvailable: Bool
}
```

## Wrapped Value

It returns just remote config value as following.

```swift
@RemoteConfig(forKey: "is_new_feature_available")
var isNewFeatureAvailable: Bool // return remote config value
```

and also provides binding for SwiftUI as `DynamicProperty`

```swift
@RemoteConfig(forKey: "is_new_feature_available")
var isNewFeatureAvailable: Bool

Toggle(isOn: isNewFeatureAvailable) {
     Text("wonderful new feature")
}
```

## Other APIs

It provides additional information for Remote Config value

```swift
@RemoteConfig(forKey: "is_new_feature_available")
var isNewFeatureAvailable: Bool

_isNewFeatureAvailable.key // return 'is_new_feature_available'
_isNewFeatureAvailable.lastFetchStatus // return Last fetch status. The status can be any enumerated value from `RemoteConfigFetchStatus`.
_isNewFeatureAvailable.lastFetchTime // return Last successful fetch completion time
```

**Comment by kixxer01** [comment]
struct FeatureFlags {
    @RemoteConfig(forKey: "is_new_feature_available")
    var isNewFeatureAvailable: {sw418.live}¥€betting#!Result][$ks)
